File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.java
class Solution {
    public int[][] solution(int n) {
        int[][] answer = new int[n][n];
        int level = 0;
        int index = 1;
        
        while (index <= n * n) {
            for (int x=level; x<n-level; x++) {
                answer[level][x] = index++;
            }
            
            for (int y=level+1; y<n-level; y++) {
                answer[y][n-level-1] = index++;
            }
            
            for (int x=n-1-level-1; x>=level; x--) {
                answer[n-1-level][x] = index++;
            }
            
            for (int y=n-1-level-1; y>level; y--) {
                answer[y][level] = index++;
            }
            
            level += 1;
        }
        return answer;
    }
}

File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.kt
/*
 * level은 채우고자 하는 테두리
 * level에 따라 각 변마다 n-level위치에서 채우면 됨.
 */
class Solution {
    fun solution(n: Int): Array<IntArray> {
        var answer: Array<IntArray> = Array(n) { IntArray(n) }
        var index = 1
        var level = 0
        
        while (index <= n * n) {
            for (x in level until n-level) {
                answer[level][x] = index++
            }
            
            for (y in level+1 until n-level) {
                answer[y][n-1-level] = index++
            }
            
            for (x in n-1-level-1 downTo level) {
                answer[n-1-level][x] = index++
            }
            
            for (y in n-1-level-1 downTo level+1) {
                answer[y][level] = index++
            }
            
            level += 1
        }
        
        return answer
    }
}